sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -50,6 +52,39 @@
 from apache_beam.utils.timestamp import Timestamp
 
 
+class CounterIncrememtingCombineFn(beam.CombineFn):
+  """CombineFn for incrementing three different counters:
+     counter, distribution, gauge,
+     at the same time concatenating words."""
+
+  def __init__(self):
+    beam.CombineFn.__init__(self)
+    self.word_counter = Metrics.counter(self.__class__, 'word_counter')
+    self.word_lengths_counter = Metrics.counter(
+        self.__class__, 'word_lengths')
+    self.word_lengths_dist = Metrics.distribution(
+        self.__class__, 'word_len_dist')
+    self.last_word_len = Metrics.gauge(self.__class__, 'last_word_len')
+
+  def create_accumulator(self):
+    return ''
+
+  def add_input(self, acc, element):
+    self.word_counter.inc(1)
+    self.word_lengths_counter.inc(len(element))
+    self.word_lengths_dist.update(len(element))
+    self.last_word_len.set(len(element))
+
+    # ''.join() converts the list to a string.
+    return ''.join(sorted(acc + element))
+
+  def merge_accumulators(self, accs):
+    return ''.join(sorted(''.join(accs)))
```

Comment:
Likewise, `return sum(accs, '')`

Reply:
sum(accs, ' ') doesn't work, because accs could be a list of letters, such as ['a', 'b', 'c'], may not be a string.